mountpoint-s3/src/main.rs

```diff
@@ -480,7 +481,7 @@ fn create_client_for_bucket(
     addressing_style: AddressingStyle,
 ) -> Result<S3CrtClient, anyhow::Error> {
     const DEFAULT_REGION: &str = "us-east-1";
-
+    let allocator = Allocator::default();
```

Comment:
Is this really necessary? Couldn't `from_region` just use the default allocator? I don't think we want to expose this outside the client.

Reply:
yes. It can. In fact, I was making that change in my work for ARN support yesterday. Changing it here as well.